api-gateway/api-console/modules/swagger/src/main/java/io/knotx/example/swagger/handler/AdminInfoHandlerFactory.java

```diff
@@ -0,0 +1,50 @@
+package io.knotx.example.swagger.handler;
+
+import io.knotx.server.api.context.ClientRequest;
+import io.knotx.server.api.context.RequestContext;
+import io.knotx.server.api.handler.RoutingHandlerFactory;
+import io.vertx.core.Handler;
+import io.vertx.reactivex.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonObject;
+import io.vertx.reactivex.core.Vertx;
+import io.vertx.reactivex.ext.web.RoutingContext;
+
+/**
+ * Primitive handler that returns either hardcoded JSON or XML based on "datatype" query parameter.
+ * It has no real applications and is provided only as a mock implementation for demonstration purposes.
+ */
+public class AdminInfoHandlerFactory implements RoutingHandlerFactory {
+
+  @Override
+  public String getName() {
+    return "admin-info";
+  }
+
+  @Override
+  public Handler<RoutingContext> create(Vertx vertx, JsonObject config) {
+    return routingContext -> {
+      RequestContext requestContext = routingContext.get(RequestContext.KEY);
+      ClientRequest clientRequest = requestContext.getRequestEvent().getClientRequest();
+
+      String type = clientRequest.getParams().get("datatype");
+      respondByType(routingContext.response(), type);
+    };
+  }
+
+  private void respondByType(HttpServerResponse response, String type) {

```

Comment:
Is it possible that response will be in yaml?

Reply:
I didn't include yaml here for two reasons
- two types are enought to demonstrate the concept
- yaml is more suited for the configuration files etc. I don't think it's commonly used for client-server configuration